RevenueCatUI/Views/PurchaseButton.swift

```diff
@@ -116,6 +106,46 @@ struct PurchaseButton: View {
 
 }
 
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, *)
+private struct PurchaseButtonLabel: View {
+
+    let package: TemplateViewConfiguration.Package
+    let colors: PaywallData.Configuration.Colors
+    let introEligibility: IntroEligibilityStatus?
+    let mode: PaywallViewMode
+
+    @Environment(\.userInterfaceIdiom)
+    private var userInterfaceIdiom
+
+    @Environment(\.isEnabled)
+    private var isEnabled
+
+    var body: some View {
+        IntroEligibilityStateView(
+            textWithNoIntroOffer: self.package.localization.callToAction,
+            textWithIntroOffer: self.package.localization.callToActionWithIntroOffer,
+            introEligibility: self.introEligibility,
+            foregroundColor: self.colors.callToActionForegroundColor
+        )
+            .frame(
+                maxWidth: self.mode.fullWidthButton
+                   ? .infinity
+                    : nil
+            )
+            .padding()
+            .padding(.vertical, self.userInterfaceIdiom == .pad ? 10 : 0)
+            .hidden(if: !self.isEnabled)
```

Comment:
Always displaying this to keep layout consistent, but hiding it when the button is disabled.